src/app/shared/chart-and-table/chart-and-table.component.ts

```diff
@@ -0,0 +1,189 @@
+import * as Highcharts from 'highcharts';
+import _sortBy from 'lodash-es/sortBy';
+import * as _ from 'lodash';
+// import chroma from 'chroma-js';
+// import * as moment from 'moment';
+// import { GRAPH_COLUMN_NAME } from '../constants/graphConstants';
+// import { DAY_DATE_FORMAT, DATE_FORMAT } from '../constants/siteConstants';
+// import { formatNumber } from '../common-utils/common-utils';
+
+@Component({
+  selector: 'app-chart-and-table',
+  templateUrl: './chart-and-table.component.html',
+  styleUrls: ['./chart-and-table.component.scss']
+})
+export class ChartAndTableComponent implements OnInit {
+
+  @Input() setChartTheme: any = [];
+  @Input() chartType: string = 'spline';
+  @Input() chartTitle: string = null;
+  @Input() chartSubTitle: string = null;
+  @Input() xAxisCategories: Array<string> = [];
+  @Input() chartSeries: Array<object> = [];
+  @Input() columnDefs: Array<object> = [];
+  @Input() tableData: Array<object> = [];
+
+  public Highcharts: typeof Highcharts = Highcharts; // Highcharts, it's Highcharts
+  public chartOptions: object = {}
+  private yAxisConfig: any = [];
+
+  // TODO:
+  private graphMeta = {};
+  private selectorGroups = [];
+  private selectors = [];
+  private config = {};
+
+  private GRAPH_SERIE_OPTIONS_DEFAULT = {
+    connectNulls: true
+  };
+
+  private GRAPH_XAXIS_OPTIONS_DEFAULT = {
+    title: {
+      text: null
+    }
+  };
+
+  private GRAPH_YAXIS_OPTIONS_DEFAULT = {
+    title: {
+      text: null
+    }
+  };
+
+  constructor() { }
+
+  ngOnInit(): void {
+  
+    this.chartInit();
+  }
+
+  private initialize() {
+
+  }
+
+  private chartInit() {
+    this.Highcharts.setOptions(this.setChartTheme);
+    this.setYAxis();
+    this.chartOptions = {
+
+      chart: {
+        type: this.chartType,
+        zoomType: 'x',
+        spacingTop: 20,
+        width: 1800,
+      },
+      title: {
+        text: this.chartTitle
+      },
+      subtitle: {
+        text: this.chartSubTitle
+      },
+      xAxis: {
+        categories: this.xAxisCategories,
+      },
+      yAxis: this.yAxisConfig,
+      tooltip: {
+        crosshairs: [true],
+        animation: true,
+        backgroundColor: '#fff',
+        borderColor: '#ddd',
+        borderRadius: 4,
+        borderWidth: 1,
+        shared: true,
+        shadow: true,
+        padding: 20,
+        style: {
+          color: '#666',
+          lineHeight: 20
+        },
+      },
+      legend: {
+        layout: 'horizontal',
+        align: 'left',
+        verticalAlign: 'top',
+        itemMarginTop: 0,
+        itemMarginBottom: 10
+      },
+
+      plotOptions: {
+        areaspline: {
+          fillOpacity: 0.09,
+        },
+        series: {
+          fillOpacity: 0.2
+        }
+      },
+      series: this.chartSeries
+    };
+  }
+
+  private setYAxis() {
+    const count = this.chartSeries.length;
+    if (count > 0) {
+      this.chartSeries.map((row, i) => {
+        const index = ((count - 1) - i);
```

Comment:
what is a reason to go in reverse order?

Reply:
Yes, to match the color of "legend" with yAxis labels to more user visibility.